include/ocpp/v201/smart_charging.hpp

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+#ifndef OCPP_V201_SMART_CHARGING_HPP
+#define OCPP_V201_SMART_CHARGING_HPP
+
+#include "ocpp/v201/enums.hpp"
+#include <limits>
+
+#include <ocpp/v201/database_handler.hpp>
+#include <ocpp/v201/evse.hpp>
+#include <ocpp/v201/ocpp_types.hpp>
+#include <ocpp/v201/transaction.hpp>
+
+namespace ocpp {
+namespace v201 {
+
+enum class ProfileValidationResultEnum {
+    Valid,
+    TxProfileMissingTransactionId,
+    TxProfileEvseIdNotGreaterThanZero,
+    TxProfileTransactionNotOnEvse,
+    TxProfileEvseHasNoActiveTransaction,
+    TxProfileConflictingStackLevel
+};
+
+/// \brief This class handles and maintains incoming ChargingProfiles and contains the logic
+/// to calculate the composite schedules
+class SmartChargingHandler {
+private:
+    std::shared_ptr<ocpp::v201::DatabaseHandler> database_handler;
+    std::vector<ChargingProfile> charging_profiles;
```

Comment:
If this field being unused is intentional, you should be able to disable the Codacy error by suppressing this check inline.
```suggestion
    // cppcheck-suppress unusedStructMember
    std::vector<ChargingProfile> charging_profiles;
```

Reply:
But what if the field is being used and Codacy is confidently wrong?